File: Official Windows Platform Sample/Universal Windows app samples/[C++]-Universal Windows app samples/Media extensions sample/C# and C++/WindowsPhone/SampleUtilities.cs
//*********************************************************
//
// Copyright (c) Microsoft. All rights reserved.
//
//*********************************************************

using System;
using Windows.UI.Core;
using Windows.UI.Xaml.Controls;

namespace SDKTemplate
{
    public partial class MainPage : Page
    {
        //
        //  Open a single file picker [with fileTypeFilter].
        //  And then, call media.SetSource(picked file).
        //  If the file is successfully opened, VideoMediaOpened() will be called and call media.Play().
        //
        public async void PickSingleFileAndSet(string[] fileTypeFilters, params MediaElement[] mediaElements)
        {
            await Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () =>
            {
                String firstFilter = fileTypeFilters[0];
                Uri uri;
                if (firstFilter.Equals(".mpg", StringComparison.OrdinalIgnoreCase))
                {
                    uri = new Uri("ms-appx:///videos/video.mpg");
                }
                else if (firstFilter.Equals(".mp4", StringComparison.OrdinalIgnoreCase))
                {
                    uri = new Uri("http://ie.microsoft.com/testdrive/Graphics/VideoFormatSupport/big_buck_bunny_trailer_480p_high.mp4");
                }
                else
                {
                    throw new ArgumentException();
                }

                for (uint i = 0; i < mediaElements.Length; ++i)
                {
                    MediaElement media = mediaElements[i];
                    media.Stop();
                    media.Source = uri;
                }
            });
        }
    }
}

File: Official Windows Platform Sample/Windows 8.1 Store app samples/99866-Windows 8.1 Store app samples/Media extensions sample/C++/WindowsPhone/SampleUtilities.cpp
//*********************************************************
//
// Copyright (c) Microsoft. All rights reserved.
//
//*********************************************************

#include "pch.h"
#include "SampleUtilities.h"

using namespace SDKSample::MediaExtensions;

using namespace Platform;

using namespace Windows::Foundation;
using namespace Windows::Foundation::Collections;
using namespace Windows::Storage;
using namespace Windows::UI::Xaml;
using namespace Windows::UI::Xaml::Controls;

using namespace concurrency;

//
//  Open a single file picker [with fileTypeFilter].
//  On the phone we just set up media elements with preconfigured urls.
//
void SampleUtilities::PickSingleFileAndSet(IVector<String^>^ fileTypeFilters, IVector<MediaElement^>^ mediaElements)
{
    String ^firstFilter = fileTypeFilters->GetAt(0);
    Uri ^uri;
    if (wcscmp(firstFilter->Data(), L".mpg") == 0)
    {
        uri = ref new Uri("ms-appx:///", "video.mpg");
    }
    else if (wcscmp(firstFilter->Data(), L".mp4") == 0)
    {
        uri = ref new Uri("http://ie.microsoft.com/testdrive/Graphics/VideoFormatSupport/big_buck_bunny_trailer_480p_high.mp4");
    }
    else
    {
        throw ref new InvalidArgumentException();
    }

    for (unsigned int i = 0; i < mediaElements->Size; ++i)
    {
        MediaElement^ media = mediaElements->GetAt(i);
        media->Stop();
        media->Source = uri;
    }
}

